sdk/core/Azure.Core/src/Pipeline/RequestMethod.cs

```
@@ -0,0 +1,102 @@
+            {
+                if (string.Equals(method, "GET", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Get;
+                }
+
+                if (string.Equals(method, "PUT", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Put;
+                }
+            }
+            else if (method.Length == 4)
+            {
+                if (string.Equals(method, "POST", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Post;
+                }
+                if (string.Equals(method, "HEAD", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Head;
+                }
+            }
+            else
+            {
+                if (string.Equals(method, "PATCH", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Patch;
+                }
+                if (string.Equals(method, "DELETE", StringComparison.OrdinalIgnoreCase))
+                {
+                    return Delete;
+                }
+            }
+
+            return new RequestMethod(method);
+        }
+
+        public bool Equals(RequestMethod other)
+        {
+            return string.Equals(Method, other.Method, StringComparison.Ordinal);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is RequestMethod other && Equals(other);
+        }
+
+        public override int GetHashCode()
+        {
+            return Method?.GetHashCode() ?? 0;
+        }
+
+        public static bool operator ==(RequestMethod left, RequestMethod right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(RequestMethod left, RequestMethod right)
+        {
+            return !left.Equals(right);
+        }
+
+        public override string ToString()
+        {
+            return Method ?? "<null>";
```

Thinking about it after your change and I wonder if this type would be easier to use with a `_method` field that might be null and a `public string Method => this._method ?? "GET";` that everything else used?  I'm also fine with what you have if you prefer this approach.

---

I think I want things to explode if the method is left uninitialized. Defaulting to GET feels like hiding an issue.

---

For what it's worth, I agree with Pavel.   A default GET would be convenient for common use, but I've seen it cause some difficult to diagnose issues when someone forgot to specify a VERB and attempted to request an update via GET.